packages/x-date-pickers/src/models/adapters.ts

```diff
@@ -0,0 +1,586 @@
+   */
+  isAfter(value: TDate, comparing: TDate): boolean;
+  /**
+   * Returns `true` if the day of the reference date is after the day of the second date.
+   * TODO v7: Consider adding a `unit` param to `isAfter` and drop this method.
+   * @template TDate
+   * @param {TDate} value The reference date.
+   * @param {TDate} comparing The date to compare with the reference date.
+   * @returns {boolean} `true` if the day of the reference date is after the day of the second date.
+   */
+  isAfterDay(value: TDate, comparing: TDate): boolean;
+  /**
+   * Returns `true` if the year of the reference date is after the year of the second date.
+   * TODO v7: Consider adding a `unit` param to `isAfter` and drop this method.
+   * @template TDate
+   * @param {TDate} value The reference date.
+   * @param {TDate} comparing The date to compare with the reference date.
+   * @returns {boolean} `true` if the year of the reference date is after the year of the second date.
+   */
+  isAfterYear(value: TDate, comparing: TDate): boolean;
+  /**
+   * Returns `true` if the reference date is before the second date.
+   * @template TDate
+   * @param {TDate} value The reference date.
+   * @param {TDate} comparing The date to compare with the reference date.
+   * @returns {boolean} `true` if the reference date is before the second date.
+   */
+  isBefore(value: TDate, comparing: TDate): boolean;
+  /**
+   * Returns `true` if the day of the reference date is before the day of the second date.
+   * TODO v7: Consider adding a `unit` param to `isBefore` and drop this method.
+   * @template TDate
+   * @param {TDate} value The reference date.
+   * @param {TDate} comparing The date to compare with the reference date.
+   * @returns {boolean} `true` if the day of the reference date is before the day of the second date.
+   */
+  isBeforeDay(value: TDate, comparing: TDate): boolean;
+  /**
+   * Returns `true` if the year of the reference date is before the year of the second date.
+   * TODO v7: Consider adding a `unit` param to `isBefore` and drop this method.
+   * @template TDate
+   * @param {TDate} value The reference date.
+   * @param {TDate} comparing The date to compare with the reference date.
+   * @returns {boolean} `true` if the year of the reference date is before the year of the second date.
+   */
+  isBeforeYear(value: TDate, comparing: TDate): boolean;
+  /**
+   * Returns `true` if the value is withing the provided range.
+   * @template TDate
+   * @param {TDate} value The value to test.
+   * @param {[TDate, TDate]} range The range in which the value should be.
+   * @returns {boolean} `true` if the value is withing the provided range.
+   */
+  isWithinRange(value: TDate, range: [TDate, TDate]): boolean;
+  /**
+   * Return the start of the year for the given date.
+   * @template TDate
+   * @param {TDate} value The original date.
+   * @returns {TDate} The start of the year.
```

Comment:
I'd suggest either going with a correct `@returns` or ditching `@returns` and using the method description only to avoid duplication. 🤔 
```suggestion
   * @returns {TDate} The given date at the start of the year.
```

Reply:
For a lot of the methods, I took the JSDoc of Date Fns (which is the inspiration for most of the methods name).

I can replace with `The start if the year of the given date`

Reply:
Gotcha. We are going to be exposing this interface publicly from now on, so, having a more appropriate JSDoc is very welcome. 👍 

> I can replace with `The start if the year of the given date`

Yup, this suggestion also sounds more correct than the current solution. 👍